### PR TITLE
Select component

### PR DESCRIPTION
**Pull request by ollips**
Task: task-531897

Link: preview-349

Adding Select component and relevant documentation.

## Testing

1. Visit [Select](https://design.docs.microsoft.com/pulls/349/components/select.html) page
2. Click through the component, ensure different states are looking good.

### CODE REVIEW

File: site/src/components/select.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+		<option value="option-3">Option 3</option>
+	</select>
+</div>
+```
+
+### Multiple select
+
+There is an option to style a multiple select dropdowns by using the `.select-multiple` class in conjunction with `.select` and adding the `multiple` attribute to the `<select>` tag.
+
+Hold down `control` button on Windows/`command` button on Mac to select multiple options.
```
Comment (homing1) [inline site/src/components/select.md:32]: suggestion: I wonder if this is better to use the actual label on the keyboard "`ctrl` key on Windows / `cmd` key on Mac".

Comment (ollips) [reply site/src/components/select.md]: Updated

File: css/src/components/form/select.scss [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+		&[disabled] {
+			border-color: $select-disabled-border-color;
+			background-color: $select-disabled-background-color;
+			color: $select-disabled-color;
+		}
+
+		@include focus-visible {
+			border-color: $select-focus-border-color;
+			outline: none !important;
```
Comment (jdanyow) [inline css/src/components/form/select.scss:56]: To ensure forced colors mode gets the same thick outline, lets use:
```
outline-width: 0.125rem; // variable
outline-style: solid;
outline-color: transparent; // forced=colors will override this with the user's preferred focus outline color.
outline-offset: 0;
```

Same change will be needed in other controls.

**Review by jdanyow — CHANGES_REQUESTED** [review]

File: css/src/components/form/select.scss (hunk 2 above)
Comment (jdanyow) [reply css/src/components/form/select.scss:56]: image

File: css/src/components/form/select.scss [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+	&.select-lg {
+		@include control-lg;
+	}
+
+	&.select-danger select {
+		border-color: $select-danger-border-color;
+
+		@include focus-visible() {
+			outline: none !important;
```
Comment (jdanyow) [inline css/src/components/form/select.scss:104]: nit- remove

File: css/src/components/form/select.scss [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+			box-shadow: $select-focus-box-shadow-size $select-danger-border-color;
+		}
+	}
+
+	&.select-success select {
+		border-color: $select-success-border-color;
+
+		@include focus-visible() {
+			outline: none !important;
```
Comment (jdanyow) [inline css/src/components/form/select.scss:113]: nit- remove

**Review by jdanyow — CHANGES_REQUESTED** [review]

File: css/src/components/form/select.scss (hunk 3 above)
Comment (ollips) [reply css/src/components/form/select.scss]: That's so disappointing 🤦🏻‍♀️Updated!

File: css/src/components/form/select.scss (hunk 4 above)
Comment (ollips) [reply css/src/components/form/select.scss]: Updated!

**Review by jdanyow — APPROVED** [review]

**Review by homing1 — APPROVED** [review]

**Pull request merged**